### PR TITLE
View tests

### PR DESCRIPTION
**Pull request by Anbranin**
Writing view tests

### CODE REVIEW

File: coverage/.last_run.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "covered_percent": 100.0
+    "covered_percent": 53.0
```
Comment (dfaulken) [inline coverage/.last_run.json]: I'll fix this, but be sure to run the whole spec suite before committing

File: spec/views/fields/edit_spec.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+include RSpecHtmlMatchers
+
+describe 'fields/edit.haml' do
+  before :each do
+    @form_draft = create :form_draft
+    @field = create :field, form_draft: @form_draft
+  end
+  it 'contains the name of the field' do
+    render
+    expect(rendered).to have_tag 'h1', with: { class: 'title' }
```
Comment (dfaulken) [inline spec/views/fields/edit_spec.rb]: This doesn't test for the name of the field, just the presence of the h1 tag which should contain it

File: spec/views/form_drafts/edit_spec.rb [hunk 3]
```diff
@@ -0,0 +1,107 @@
+require 'rails_helper'
+include RSpecHtmlMatchers
+
+describe 'form_drafts/edit.haml' do
+  before :each do
+    @draft = create :form_draft
+    @top_field = create :field, form_draft: @draft
```
Comment (dfaulken) [inline spec/views/form_drafts/edit_spec.rb]: I'd prefer to align the `=` of lines 7-9, to make it obvious that they're all the same invocation.

File: spec/views/form_drafts/edit_spec.rb [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    @field = create :field, form_draft: @draft
+    @bottom_field = create :field, form_draft: @draft
+  end
+  it 'contains the name of the form draft' do
+    render
+    expect(rendered).to have_tag 'h1', with: { class: 'title' }
```
Comment (dfaulken) [inline spec/views/form_drafts/edit_spec.rb]: Same here as above, doesn't test for the name.

File: spec/views/form_drafts/edit_spec.rb [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+  end
+  it 'has a form to edit the form draft' do
+    render
+    action_path = form_draft_path @draft
+    expect(rendered).to have_form action_path, :post
```
Comment (dfaulken) [inline spec/views/form_drafts/edit_spec.rb]: I'd also like to see a test about the attributes of the form draft which can be edited (name).

File: spec/views/form_drafts/edit_spec.rb [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+  end
+  it 'has inputs for each field' do
+    render
+    expect(rendered).to have_tag 'tr' do
+      with_hidden_field 'form_draft[fields_attributes][0][number]'
```
Comment (dfaulken) [inline spec/views/form_drafts/edit_spec.rb]: I'd like to align the arguments to these matchers.

File: spec/views/form_drafts/edit_spec.rb [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+      with_checkbox 'form_draft[fields_attributes][0][required]'
+    end
+  end
+  context 'input data type is options' do
+    before :each do
+      @field.update data_type: 'options', options: %w(car van)
+    end
+    it 'lists the current options' do
+      render
+      expect(rendered).to have_tag 'li'
```
Comment (dfaulken) [inline spec/views/form_drafts/edit_spec.rb]: Here again, I'm also concerned with the text in the li, not just its presence.

**Comment by dfaulken** [comment]
Good start, some fixes to do before moving on to another view.

File: spec/views/fields/edit_spec.rb (hunk 2 above)
Comment (Anbranin) [inline spec/views/fields/edit_spec.rb]: rewrote this

File: spec/views/form_drafts/edit_spec.rb (hunk 3 above)
Comment (Anbranin) [inline spec/views/form_drafts/edit_spec.rb]: done

**Pull request merged**